keps/sig-cli/2227-kubectl-default-container/README.md

```diff
@@ -0,0 +1,300 @@
+
+Also this KEP proposes the change of behavior of kubectl commands that relies on the --container flag to read and make use of this annotation when the flag is not provided.
+
+We don't have a general default container name attribute for pod, and this would change pod spec and is not acceptable. Why not use `kubectl.kubernetes.io/default-container` as the default container name annotation for all four commands above?
+
+### Current CLI Behaviors
+
+The following is the behavior of kubectl commands that can specify a container with --container:
+
+- `kubectl attach`, `kubectl cp` and `kubectl exec`
+  The three above have similar behavior: if --container flag is omitted, the first container of the Pod will be chosen.
+
+- `kubectl logs`
+  There's no default value. If a Pod have multiple containers, the operator needs to select which container to show the logs, or use the flag `--all-containers`
+  Also, there's support for the annotation kubectl.kubernetes.io/default-logs-container in the Pod, that specified the default container to show the log.
+
+- `kubectl debug`
+  Use this option to specify container name for debug container. 
+
+### User Stories
+
+User story 1: Julia, the operator of a deployment called "backend" that generates some stack traces locally. Her environment uses Service Mesh, and because of this, every time she tries to copy those stack traces she gets an error because the file does not exist, and this happens because kubectl cp points to the service mesh sidecar as the first container.
+
+Story 2: John, a developer of a PHP application always needs to run a command inside the application Pod. Because of the way John structured the Pod (a Container with NGINX and the other with php_fpm) every time he wants to execute this command on the php_fpm but he forgets to use the flag --container, and this way the command gets executed in the wrong container.
+
+### Proposal Details
+
+A single, generic and well known annotation for all above commands like `kubectl.kubernetes.io/default-container` is a good choice to avoid needing many new annotations in the future.
+
+There are currently 3 commands that consume this annotation if `--container` is not specified.
+- `kubectl exec`
+- `kubectl attach`
+- `kubectl cp`
+- `kubectl logs`
+
+However, there is an exception that `kubectl logs` will consume this annotation only if no `--all-containers` option is specified. If `--all-containers` is specified, all pods logs should be returned.
+
+`kubectl debug` will not consume this annotation, as `--container` for `kubectl debug` is to speicify new debug container name to use. It is quite differenet with the annotation meaning here.
+
+If `kubectl.kubernetes.io/default-logs-container` is specified, we should use this annotation instead of the general one for `kubectl log` and use general annotation for other commands.
```

Comment:
Add information that we will warn users that `kubectl.kubernetes.io/default-logs-container` is deprecated and they should use `kubectl.kubernetes.io/default-container` instead.

Reply:
About this, I've questioned if we really want to deprecate that or only prioritize :) I guess, it would be good to have both of them working and one not impacting in the other :)

Reply:
This may depend on whether we should support different default container for logs and exec.
- If not, deprecating the old annotation is better way forward.
- If yes, we should keep them all.

Reply:
As @soltysh already pointed before, maybe it would be good to add here the information that we will warn users about default-logs-container annotation deprecation

Reply:
Yes, especially add here information that we'll keep the old annotation working until 1.25 as discussed yesterday on slack.